diff --git a/config/dev.exs b/config/dev.exs
index ab3e83c12..a8a86c724 100644
--- a/config/dev.exs
+++ b/config/dev.exs
@@ -20,6 +20,8 @@
 
 config :pleroma, Pleroma.Emails.Mailer, adapter: Swoosh.Adapters.Local
 
+config :pleroma, :http_security, enabled: false
+
 # ## SSL Support
 #
 # In order to use HTTPS in development, a self-signed
diff --git a/lib/pleroma/http/adapter_helper/gun.ex b/lib/pleroma/http/adapter_helper/gun.ex
index 74ab9851e..c571b2aee 100644
--- a/lib/pleroma/http/adapter_helper/gun.ex
+++ b/lib/pleroma/http/adapter_helper/gun.ex
@@ -37,7 +37,7 @@ def options(incoming_opts \\ [], %URI{} = uri) do
   defp add_scheme_opts(opts, %{scheme: "http"}), do: opts
 
   defp add_scheme_opts(opts, %{scheme: "https"}) do
-    Keyword.put(opts, :certificates_verification, true)
+    Keyword.put(opts, :certificates_verification, false)
   end
 
   defp put_timeout(opts) do
diff --git a/lib/pleroma/http/adapter_helper/hackney.ex b/lib/pleroma/http/adapter_helper/hackney.ex
index f3be1f3d0..5fae2dae5 100644
--- a/lib/pleroma/http/adapter_helper/hackney.ex
+++ b/lib/pleroma/http/adapter_helper/hackney.ex
@@ -3,11 +3,14 @@
 # SPDX-License-Identifier: AGPL-3.0-only
 
 defmodule Pleroma.HTTP.AdapterHelper.Hackney do
+  require Logger
+
   @behaviour Pleroma.HTTP.AdapterHelper
 
   @defaults [
     follow_redirect: true,
-    force_redirect: true
+    force_redirect: true,
+    insecure: true
   ]
 
   @spec options(keyword(), URI.t()) :: keyword()
